calendar/calendar-tool/tool/src/java/org/sakaiproject/calendar/entityproviders/CalendarEventEntityProvider.java

```diff
@@ -338,7 +325,7 @@ private List getMergedCalendarEventsForSite(final String siteId, final TimeRange
 			CalendarEvent event = (CalendarEvent) o;
 
 			CalendarEventDetails eventDetails = new CalendarEventDetails(event);
-			eventDetails.setEventImage(eventImageMap.get(event.getType()));
+			eventDetails.setEventIcon(eventIconMap.get(event.getType()));
```

Comment:
This could fail if the `eventIconMap` hasn't first been setup by the `getCalendarEventsForSite()` method.